apps/authorization/test_data_access_grant.py

```diff
@@ -207,3 +207,64 @@ def test_check_and_update_grants(self):
             checks['unique_tokens'],
             checks['grants'],
         )
+
+    def test_permission_deny_on_app_that_is_disabled(self):
+        '''
+        BB2-149 leverage application.active, user.is_active to deny permission
+        to an application or applications under a user (organization)
+        '''
+        redirect_uri = 'http://localhost'
+        # create a user
+        user = self._create_user('anna', '123456')
+        capability_a = self._create_capability('Capability A', [])
+        # create an application and add capabilities
+        application = self._create_application(
+            'an app',
+            grant_type=Application.GRANT_AUTHORIZATION_CODE,
+            user=user,
+            redirect_uris=redirect_uri)
+        application.scope.add(capability_a)
+        application.active = False
+        application.save()
+        # user logs in
+        self.client.login(username='anna', password='123456')
+
+        payload = {
+            'client_id': application.client_id,
+            'response_type': 'code',
+            'redirect_uri': redirect_uri,
+        }
+        response = self.client.get('/v1/o/authorize', data=payload)
+        payload = {
+            'client_id': application.client_id,
+            'response_type': 'code',
+            'redirect_uri': redirect_uri,
+            'scope': ['capability-a'],
+            'expires_in': 86400,
+            'allow': True,
+        }
+
+        try:
+            self.client.post(response['Location'], data=payload)
+        except PermissionDenied as permErr:
+            errStr = str(permErr)
+            errwords = errStr.split()
+            packedErrStr = "-".join(errwords)
+            msgwords = settings.APPLICATION_TEMPORARILY_INACTIVE.split()
+            packedMsg = "-".join(msgwords)
+            packedMsg = packedMsg.format("an-app")
+            self.assertEqual(packedErrStr, packedMsg)
+
+        try:
+            self.client.post(response['Location'], data=payload)
+        except PermissionDenied as permErr:
+            errStr = str(permErr)
+            errwords = errStr.split()
+            packedErrStr = "-".join(errwords)
+            msgwords = settings.APPLICATION_TEMPORARILY_INACTIVE.split()
+            packedMsg = "-".join(msgwords)
+            packedMsg = packedMsg.format("an-app")
+            self.assertEqual(packedErrStr, packedMsg)
```

Comment:
@JFU-GIT Found that this test will still pass, if `application.active = True` is set previously instead of `False`.

This section could be replaced with the following code:
```suggestion
        msg_expected = settings.APPLICATION_TEMPORARILY_INACTIVE.format("an app")
        with self.assertRaises(PermissionDenied) as cm:
            self.client.post(response['Location'], data=payload)
        self.assertEqual(str(cm.exception), msg_expected)

```

Reply:
this is cleaner